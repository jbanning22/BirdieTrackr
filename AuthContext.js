import {createContext} from 'react';

export const AuthContext = createContext();

// export const AuthProvider = ({children}) => {
//   const [signedIn, setSignedIn] = useState(false);

//   return (
//     <AuthContext.Provider value={{signedIn, setSignedIn}}>
//       {children}
//     </AuthContext.Provider>
//   );
// };
