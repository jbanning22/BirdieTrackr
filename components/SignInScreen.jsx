import {
  StyleSheet,
  Text,
  View,
  SafeAreaView,
  TouchableOpacity,
  TextInput,
  Button,
  ScrollView,
  KeyboardAvoidingView,
} from 'react-native';
import React, {useState, useContext} from 'react';
import axios from 'axios';
import AsyncStorage from '@react-native-async-storage/async-storage';
import {useNavigation, CommonActions} from '@react-navigation/native';
import {AuthContext} from '../AuthContext';

const SignInScreen = ({navigation}) => {
  const {signedIn, setSignedIn} = useContext(AuthContext);
  const [formData, setFormData] = useState({
    email: '',
    password: '',
  });
  const [validationErrors, setValidationErrors] = useState({});
  //   const [email, setEmail] = useState('');
  //   const [password, setPassword] = useState('');

  const signIn = async () => {
    const errorsExist = Object.values(validationErrors).some(
      error => error !== '',
    );
    if (errorsExist) {
      console.log('Please fix the form errors before signing up.');
      return;
    } else {
      try {
        const signInRes = await axios.post(
          'http://192.168.1.154:3000/auth/signin',
          {
            email: formData.email,
            password: formData.password,
          },
        );
        if (signInRes.status === 200) {
          const access_token = await signInRes.data.access_token;
          const refresh_token = await signInRes.data.refresh_token;

          await AsyncStorage.setItem('token', access_token);
          await AsyncStorage.setItem('ReToken', refresh_token);
          setSignedIn(true);
          navigation.navigate('App', {screen: 'Scorecard'});
          return signInRes.data;
        }
      } catch (error) {
        console.log('error signing in', error);
      }
    }
  };
  const validateEmail = () => {
    const isValid = /\S+@\S+\.\S+/.test(formData.email);
    setValidationErrors({
      ...validationErrors,
      email: isValid ? '' : 'Email is required and must be valid.',
    });
  };

  return (
    // <SafeAreaView>
    <ScrollView contentContainerStyle={styles.box1}>
      <Text style={styles.homeText}>Welcome Back!</Text>
      <KeyboardAvoidingView>
        <TextInput
          placeholder="Email"
          style={
            validationErrors.email
              ? styles.loginTextInput2
              : styles.loginTextInput1
          }
          value={formData.email}
          onChangeText={text => setFormData({...formData, email: text})}
          //   value={email}
          //   onChangeText={setEmail}
          onBlur={validateEmail}
          autoCorrect={false}
          autoCapitalize={'none'}
        />
        <Text style={{color: 'red'}}>{validationErrors.email}</Text>
        <TextInput
          placeholder="Password"
          style={styles.loginTextInput1}
          value={formData.password}
          onChangeText={text => setFormData({...formData, password: text})}
          //   value={password}
          //   onChangeText={setPassword}
          autoCapitalize={'none'}
          secureTextEntry={true}
          autoCorrect={false}
        />
      </KeyboardAvoidingView>
      <TouchableOpacity style={styles.loginButton} onPress={signIn}>
        <Text style={styles.textButton}>Log in</Text>
      </TouchableOpacity>
      <Button title="Back" onPress={() => navigation.navigate('Landing')} />
    </ScrollView>
    // {/* </SafeAreaView> */}
  );
};

export default SignInScreen;

const styles = StyleSheet.create({
  box1: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  homeText: {
    fontSize: 40,
    fontWeight: '500',
    fontFamily: 'Helvetica',
    marginBottom: 80,
  },
  loginButton: {
    width: 200,
    height: 50,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#52BEDB',
    marginBottom: 40,
    marginTop: 20,
  },
  loginTextInput2: {
    height: 50,
    width: 200,
    padding: 10,
    backgroundColor: '#F9908D',
    marginTop: 10,
  },
  loginTextInput1: {
    height: 50,
    width: 200,
    padding: 10,
    backgroundColor: 'white',
    marginTop: 10,
  },
  textButton: {
    fontSize: 18,
    color: 'white',
  },
});
