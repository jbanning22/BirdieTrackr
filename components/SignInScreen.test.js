import React from 'react';
import {render, screen} from '@testing-library/react-native';
import SignInScreen from './SignInScreen';
import {AuthContext} from '../AuthContext';

jest.mock('@fortawesome/react-native-fontawesome', () => ({
  FontAwesomeIcon: () => null,
}));

describe('Sign In Screen', () => {
  it('should match the snapshot', () => {
    const authContextValue = {
      signedIn: false,
      setSignedIn: jest.fn(),
    };
    const {toJSON} = render(
      <AuthContext.Provider value={authContextValue}>
        <SignInScreen />
      </AuthContext.Provider>,
    );
    expect(toJSON()).toMatchSnapshot();
  });
  it('should render title', async () => {
    const authContextValue = {
      signedIn: false,
      setSignedIn: jest.fn(),
    };
    render(
      <AuthContext.Provider value={authContextValue}>
        <SignInScreen />
      </AuthContext.Provider>,
    );
    expect(screen.getByText('Sign In to BirdieTrackr')).toBeTruthy();
  });
  it('should display log in button', async () => {
    const authContextValue = {
      signedIn: false,
      setSignedIn: jest.fn(),
    };
    render(
      <AuthContext.Provider value={authContextValue}>
        <SignInScreen />
      </AuthContext.Provider>,
    );
    expect(screen.getByText('Sign in')).toBeTruthy();
  });
  it('should display forgot password', async () => {
    const authContextValue = {
      signedIn: false,
      setSignedIn: jest.fn(),
    };
    render(
      <AuthContext.Provider value={authContextValue}>
        <SignInScreen />
      </AuthContext.Provider>,
    );
    expect(screen.getByText('Forgot Password?')).toBeTruthy();
  });
  it('should display legal text', async () => {
    const authContextValue = {
      signedIn: false,
      setSignedIn: jest.fn(),
    };
    const {getByText} = render(
      <AuthContext.Provider value={authContextValue}>
        <SignInScreen />
      </AuthContext.Provider>,
    );
    const text1 = getByText('Privacy Policy');
    const text2 = getByText('BirdieTrackr Terms & Conditions');

    expect(text1).toBeDefined();
    expect(text2).toBeDefined();
  });
});
