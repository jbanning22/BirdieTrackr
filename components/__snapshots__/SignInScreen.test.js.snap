// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign In Screen should match the snapshot 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignItems": "center",
    }
  }
>
  <View
    style={
      {
        "alignSelf": "flex-start",
        "marginBottom": 100,
        "marginLeft": 20,
        "marginTop": 20,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    />
  </View>
  <Text
    style={
      {
        "fontFamily": "Satoshi-Medium",
        "fontSize": 28,
        "fontWeight": "600",
        "marginBottom": 80,
      }
    }
  >
    Sign In to 
    <Text
      style={
        {
          "color": "#45B369",
        }
      }
    >
      BirdieTrackr
    </Text>
  </Text>
  <View
    onLayout={[Function]}
    style={
      {
        "paddingBottom": 0,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "white",
          "borderRadius": 15,
          "flexDirection": "row",
          "height": 56,
          "width": 327,
        }
      }
    >
      <TextInput
        autoCapitalize="none"
        autoCorrect={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        placeholder="Email"
        style={
          {
            "alignSelf": "center",
            "flex": 1,
            "fontFamily": "Satoshi-Medium",
            "height": 60,
            "marginBottom": 8,
            "marginLeft": 10,
            "marginTop": 10,
            "padding": 10,
            "width": 327,
          }
        }
        value=""
      />
    </View>
    <Text
      style={
        {
          "color": "red",
          "padding": 5,
        }
      }
    />
    <View
      style={
        {
          "backgroundColor": "white",
          "borderRadius": 15,
          "flexDirection": "row",
          "height": 56,
          "width": 327,
        }
      }
    >
      <TextInput
        autoCapitalize="none"
        autoCorrect={false}
        onChangeText={[Function]}
        placeholder="Password"
        secureTextEntry={true}
        style={
          {
            "alignSelf": "center",
            "flex": 1,
            "fontFamily": "Satoshi-Medium",
            "height": 60,
            "marginBottom": 8,
            "marginLeft": 10,
            "marginTop": 10,
            "padding": 10,
            "width": 327,
          }
        }
        value=""
      />
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      {
        "margin": 15,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          {
            "fontFamily": "Satoshi-Medium",
          }
        }
      >
        Forgot Password?
      </Text>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#2D6061",
        "borderRadius": 12,
        "fontFamily": "Satoshi-Medium",
        "height": 60,
        "justifyContent": "center",
        "marginBottom": 40,
        "opacity": 1,
        "width": 327,
      }
    }
  >
    <Text
      style={
        {
          "color": "white",
          "fontFamily": "Satoshi-Medium",
          "fontSize": 18,
          "fontWeight": "500",
        }
      }
    >
      Sign in
    </Text>
  </View>
  <View>
    <Text
      numberOfLines={2}
      style={
        {
          "color": "#6B7280",
          "fontFamily": "Satoshi-Medium",
          "fontSize": 14,
          "marginTop": 150,
          "paddingBottom": 5,
          "paddingHorizontal": 16,
          "textAlign": "center",
        }
      }
    >
      By signing in, you agree to the
       
      <Text
        style={
          {
            "color": "black",
            "fontFamily": "Satoshi-Medium",
            "fontWeight": "400",
          }
        }
      >
        BirdieTrackr Terms & Conditions
      </Text>
       
      and 
      <Text
        style={
          {
            "color": "black",
            "fontFamily": "Satoshi-Medium",
            "fontWeight": "400",
          }
        }
      >
        Privacy Policy
      </Text>
      .
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "marginTop": 22,
      }
    }
  >
    <Modal
      animationType="slide"
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      transparent={true}
      visible={false}
    />
  </View>
</RCTSafeAreaView>
`;
